### PR TITLE
Objectfs - Save extra query when manipulator class updates and object…

### PR DESCRIPTION
**Pull request by gleimermora**
… record #213

### CODE REVIEW

File: classes/local/object_manipulator/checker.php [hunk 1]
```diff
@@ -50,12 +50,16 @@ protected function get_query_name() {
     }
 
     protected function get_candidates_sql() {
-        $sql = 'SELECT f.contenthash
+        $sql = 'SELECT f.contenthash,
```
Comment (brendanheywood) [inline classes/local/object_manipulator/checker.php]: purely for consistency I think the column order here should match the others ie id, hash then size

File: lib.php [hunk 2]
```diff
@@ -36,36 +36,55 @@
...
-    global $DB;
 
-    $newobject = new \stdClass();
+/**
+ * @param string $contenthash
+ * @param int $newlocation
+ * @return stdClass
+ * @throws dml_exception
+ */
+function update_object_record($contenthash, $newlocation) {
```
Comment (brendanheywood) [inline lib.php]: this is good but I found the name a little misleading, as I would have expected this to be the one which accepted an object. How about instead 'update_object_by_hash'

File: classes/local/object_manipulator/checker.php (hunk 1 above)
Comment (gleimermora) [reply classes/local/object_manipulator/checker.php]: updated

File: lib.php (hunk 2 above)
Comment (gleimermora) [reply lib.php]: updated to update_object_by_hash

File: classes/local/object_manipulator/manipulator.php [hunk 3]
```diff
@@ -25,6 +25,8 @@
 
 namespace tool_objectfs\local\object_manipulator;
 
+use dml_exception;
```
Comment (Peterburnett) [inline classes/local/object_manipulator/manipulator.php]: This doesnt seem to be necessary. Mightve been auto pulled in by the IDE

File: tests/tool_objectfs_testcase.php [hunk 4]
```diff
@@ -18,6 +18,9 @@
 
 defined('MOODLE_INTERNAL') || die();
 
+use dml_exception;
```
Comment (Peterburnett) [inline tests/tool_objectfs_testcase.php]: These also probably shouldnt be here

File: classes/local/object_manipulator/checker.php [hunk 5]
```diff
@@ -49,15 +49,22 @@ protected function get_query_name() {
         return 'get_check_candidates';
     }
 
+    /**
+     * @return string
+     */
     protected function get_candidates_sql() {
-        $sql = 'SELECT f.contenthash
+        return 'SELECT MAX(f.id) AS fid,
```
Comment (Peterburnett) [inline classes/local/object_manipulator/checker.php]: After grepping the Codebase, the fid is never used, as you have swapped over to o.id as the primary identifier. Is the fid still needed? I could be missing something

File: classes/local/object_manipulator/manipulator.php (hunk 3 above)
Comment (gleimermora) [reply classes/local/object_manipulator/manipulator.php]: it just looks a bit cleaner with the `use` statement. :)

File: classes/local/object_manipulator/checker.php (hunk 5 above)
Comment (gleimermora) [reply classes/local/object_manipulator/checker.php]: `fid` is not referenced, I just want to keep it for consistency.

**Pull request merged**